source/languageDetection.py

```diff
@@ -0,0 +1,284 @@
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2015 Dinesh Kaushal, NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+
+
+"""Handles detection of languages in Unicode text.
+"""
+
+import speech
+from speech import SpeechCommand
+from speech import LangChangeCommand
+import languageHandler
+import config
+from unicodeScriptData import scriptCode
+from collections import OrderedDict
+from logHandler import log
+import unicodedata
+
+# maintains list of priority languages as a list of languageID, ScriptName, and LanguageDescription
+languagePriorityListSpec = []
+
+scriptIDToLangID = {}
+
+def initialize():
+	#reverse of langIDToScriptID, required to obtain language id for a specific script 
+	for languageID in langIDToScriptID:
+		langIDToScriptID[languageID] = (langIDToScriptID[languageID] + ('Number',)) if isinstance(langIDToScriptID[languageID], tuple) else (langIDToScriptID[languageID], 'Number')
```

Comment:
this code could be simplified by ensuring that all items in the `langIDToScriptID` dict are tuples. This would be my preference.

Although, why tuples? Wouldn't a list make more sense in this case?

Reply:
simplified. tuples are used as we don't need to change these values after initialization.